easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,116 @@
+        except OSError as err:
+            self.log.error("Failed to symlink %s to %s: %s" % (orig, dest, err))
+
+        # extract some dependencies that we really need and can't find anywhere else
+        # alglib tarball has version in name, so lets find it with a glob
+        alglib_tar = glob.glob(os.path.join(os.getcwd(), 'external', 'alglib*.tgz'))[0]
+        for src in ['bilib.tgz', 'bilib.tgz', 'condor.tgz', alglib_tar]:
+            external_path = os.path.join(os.getcwd(), 'external')
+            extract_file(os.path.join(external_path, src), external_path)
+
+        cmd = ('{preconfigopts} PATH=$PWD:$PATH xmipp_python external/scons/scons.py mode=configure -j {parallel}'
+               ' --config=force  profile=no fast=yes warn=no release=yes gtest=no cuda=no debug=no matlab=no'
+               ' java=no LINKERFORPROGRAMS="$CXX" MPI_BINDIR="$EBROOTIMPI" JAVA_HOME="$JAVA_HOME" JAVAC=javac CC="$CC"'
+               ' CXXFLAGS="$CXXFLAGS -DMPICH_IGNORE_CXX_SEEK -I$EBROOTPYTHON/include/python{short_python_ver}"'
+               ' CXX="$CXX" MPI_CC="$MPICC" MPI_CXX="$MPICXX" MPI_INCLUDE="$MPI_INC_DIR" MPI_LIBDIR="$MPI_LIB_DIR"'
+               ' MPI_LINKERFORPROGRAMS="$MPICC" LIBPATH="$LD_LIBRARY_PATH" {configopts}').format(
+                   parallel=self.cfg['parallel'], short_python_ver=self.python_short_ver,
+                   preconfigopts=self.cfg['preconfigopts'], configopts=self.cfg['configopts']
+                   )
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def build_step(self):
+        """Custom build procedure for Xmipp: call the scons wrapper with compile argument"""
+        cmd = 'LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/lib PATH=$PWD:$PATH xmipp_python external/scons/scons.py' \
+              'mode=compile -j {parallel}'.format(parallel=self.cfg['parallel'])
+
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def install_step(self):
+        """install step for xmipp, this builds a local database and seems to do some tests?"""
+        cmd = 'XMIPP_HOME=$PWD PATH=$PWD:$PWD/bin:$PATH PYTHONPATH=$PYTHONPATH:$PWD/protocols:' \
+              '$PWD/libraries/bindings/python/:$EBROOTPYTHON/lib/python{short_python_ver}/lib-dynload/' \
+              ' python setup.py install'.format(self.python_short_ver)
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for GAMESS-US."""
+        custom_paths = {
+            'files': ['xmipp_%s' % x for x in ['imagej', 'mpi_run', 'phantom_create', 'tomo_project', 'volume_align']],
+            'dirs': ['lib/'],
+        }
+        super(self.__class__, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self):
+        """Define GAMESS-US specific variables in generated module file, i.e. $GAMESSUSROOT."""
+        txt = super(self.__class__, self).make_module_extra()
+        txt += self.module_generator.set_environment('XMIPP_HOME', '$root')
```

Comment:
this will not work with module files in Lua syntax

@pforai: do we have an alternative in the works for this in your PR https://github.com/hpcugent/easybuild-framework/pull/1060?